### PR TITLE
TM-103: Implemented the count of the total number of vocabularies

### PR DESCRIPTION
**Pull request by janoliver20**

### CODE REVIEW

File: be/src/services/vocabulary.service.ts [hunk 1]
```diff
@@ -53,14 +57,21 @@ export class VocabularyService {
...
-        return {
-            offset: query.offset,
-            rows: dbos.length,
-            totalItems: 0, // TODO
-            items: dbos
-        };
+        return Promise.all([
+            // @ts-ignore
+            this.collection.find(filter, options).toArray(),
+            this.countCollectionItems()
```
Comment (ja-fra) [inline be/src/services/vocabulary.service.ts]: This does not provide the correct number: It not about _all_ the vocabularies, but those that match the query/filter...

**Review by ja-fra — CHANGES_REQUESTED** [review]

File: be/src/services/entity.service.ts [hunk 2]
```diff
@@ -130,10 +130,11 @@ export class EntityService {
     const { options, filter } = this.transformToMongoDBFilterOption(query, id);
     // @ts-ignore
     const dbos: Entity[] = (await EntityService.collection().find(filter, options).toArray()) as Entity[];
+    const size = dbos.length
     return {
       offset: query.offset,
       rows: dbos.length,
-      totalItems: 0, // TODO
+      totalItems: size, // TODO
```
Comment (ManuelTS) [inline be/src/services/entity.service.ts:137]: 1. Remove the `TODO` comment
2. Using a variable with the same content as the above row is false, then we would not net totalItems at all. Instead:
3. We need the information of how many items are in total without any paging - no consideration of offset nor rows - are found for the filter with options on the DB. Is that correct @ja-fra ?

**Review by ManuelTS — CHANGES_REQUESTED** [review]

File: be/src/services/vocabulary.service.ts [hunk 3]
```diff
@@ -116,10 +116,11 @@ export class VocabularyService {
     const { options, filter } = this.transformToMongoDBFilterOption(query);
     // @ts-ignore
     const dbos: Vocabulary[] = (await VocabularyService.collection().find(filter, options).toArray()) as Vocabulary[];
+    const size = dbos.length
     return {
       offset: query.offset,
       rows: dbos.length,
-      totalItems: 0, // TODO
+      totalItems: size, // TODO
```
Comment (ManuelTS) [inline be/src/services/vocabulary.service.ts]: 1. Remove the `TODO`comment
2. From my point of view, using the variable with the same content as the row above is wrong. Instead:
3. We need here the total items of the filter + options without any paging applied. @ja-fra, do I understand this correctly?

**Review by ManuelTS — CHANGES_REQUESTED** [review]

**Review by ManuelTS — APPROVED** [review]

**Review by ManuelTS — APPROVED** [review]
Please fix the lint errors, but already approved

**Pull request merged**